In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

4. Using the pandas read_csv() method, read the GDP dataset into your notebook as a DataFrame called gdp_df. Take a look at the first few and last few rows to familiarize yourself with what is contained in this dataset.


In [2]:
gdp_df = pd.read_csv('../data/gdp_percapita.csv')

In [3]:
gdp_df.head()

Country or Area  Year        Value  Value Footnotes
0     Afghanistan  2021  1517.016266              NaN
1     Afghanistan  2020  1968.341002              NaN
2     Afghanistan  2019  2079.921861              NaN
3     Afghanistan  2018  2060.698973              NaN
4     Afghanistan  2017  2096.093111              NaN

In [4]:
gdp_df.tail()

Country or Area  Year        Value  Value Footnotes
7657        Zimbabwe  1994  2670.106615              NaN
7658        Zimbabwe  1993  2458.783255              NaN
7659        Zimbabwe  1992  2468.278257              NaN
7660        Zimbabwe  1991  2781.787843              NaN
7661        Zimbabwe  1990  2704.757299              NaN

5. How many rows and columns are in gdp_df? What are the data types of each column?

In [5]:
gdp_df.shape


(7662, 4)

In [6]:
gdp_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7662 entries, 0 to 7661
Data columns (total 4 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Country or Area  7662 non-null   object 
 1   Year             7662 non-null   int64  
 2   Value            7662 non-null   float64
 3   Value Footnotes  0 non-null      float64
dtypes: float64(2), int64(1), object(1)
memory usage: 239.6+ KB


Answers: 7662 rows & 4 columns - object, int64, float64, float64

6. Drop the Value Footnotes column and rename the remaining three to 'Country', 'Year', and 'GDP_Per_Capita'.

In [7]:
gdp_df = gdp_df.rename(columns={"Country or Area":"Country", "Value":"GDP_Per_Capita"})

In [8]:
gdp_df = gdp_df.drop("Value Footnotes", axis=1)

In [9]:
print(gdp_df.head())

       Country  Year  GDP_Per_Capita
0  Afghanistan  2021     1517.016266
1  Afghanistan  2020     1968.341002
2  Afghanistan  2019     2079.921861
3  Afghanistan  2018     2060.698973
4  Afghanistan  2017     2096.093111


7. How many countries have data for all years? Which countries are missing many years of data? Look at the number of observations per year. What do you notice?

Answer: 
    

In [10]:
unique_years = gdp_df["Year"].unique()

In [11]:
print(unique_years)

[2021 2020 2019 2018 2017 2016 2015 2014 2013 2012 2011 2010 2009 2008
 2007 2006 2005 2004 2003 2002 2022 2001 2000 1999 1998 1997 1996 1995
 1994 1993 1992 1991 1990]


In [12]:
count_years = gdp_df.groupby("Country").size().reset_index(name="Count")

In [13]:
print(count_years)

                         Country  Count
0                    Afghanistan     20
1    Africa Eastern and Southern     33
2     Africa Western and Central     33
3                        Albania     33
4                        Algeria     33
..                           ...    ...
237                     Viet Nam     33
238           West Bank and Gaza     29
239                        World     33
240                       Zambia     33
241                     Zimbabwe     33

[242 rows x 2 columns]


In [14]:
equal_33 = count_years[count_years.Count == 33]

In [15]:
not_33 = count_years[count_years.Count != 33]

In [16]:
not_33.sort_values("Count")

Country  Count
53                    Djibouti     10
197                    Somalia     10
225   Turks and Caicos Islands     12
192  Sint Maarten (Dutch part)     14
112                     Kosovo     15
36              Cayman Islands     16
154                      Nauru     19
0                  Afghanistan     20
47                    Curaçao     22
213    São Tomé and Principe     22
169                      Palau     22
166         Other small states     23
180                      Qatar     23
219                Timor-Leste     23
124                    Liberia     23
125                      Libya     24
185                 San Marino     25
149                 Montenegro     26
195               Small states     26
194                   Slovenia     28
188                     Serbia     28
126                  Lithuania     28
135                   Maldives     28
120                     Latvia     28
46                     Croatia     28
103                     Israel     28
97                     Iceland     28
147                    Moldova     28
64                     Estonia     28
24      Bosnia and Herzegovina     29
238         West Bank and Gaza     29
32                    Cambodia     30
113                     Kuwait     31
224               Turkmenistan     31
193            Slovak Republic     31
122                    Lebanon     32
22                      Bhutan     32
10                       Aruba     32
221                      Tonga     32
91                     Hungary     32